main.nf

```diff
@@ -2,83 +2,74 @@
 
 // Workflow to perform k-fold cross validation over the imput dataset
 
-//cross validation
-if(params.run == "True"){
-	// generate folds of cell indexes
-	
 K_RANGE = Channel.from(1..params.generate_folds.k_folds_num).map{it -> "Fold_$it"} 
+MATRIX_CH = Channel.fromPath(params.matrix).first()
```

Comment:
I found it necessary to add .first() here- this makes sure these are value channels and can be re-used.